internal/commands/app/create_test.go

```diff
@@ -230,222 +230,250 @@ func TestAppCreateHandler(t *testing.T) {
+				},
+			},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+			defer teardown()
+			profile.SetRealmBaseURL("http://localhost:8080")
+
+			out, ui := mock.NewUI()
+
+			var createdApp realm.App
+			rc := mock.RealmClient{}
+			rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+				createdApp = realm.App{
+					GroupID:     groupID,
+					ID:          "456",
+					ClientAppID: name + "-abcde",
+					Name:        name,
+					AppMeta:     meta,
+				}
+				return createdApp, nil
+			}
 
-		out, ui := mock.NewUI()
+			var importAppData interface{}
+			rc.ImportFn = func(groupID, appID string, appData interface{}) error {
+				importAppData = appData
+				return nil
+			}
 
-		var createdApp realm.App
-		rc := mock.RealmClient{}
-		rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
-			createdApp = realm.App{
-				GroupID:     groupID,
-				ID:          "456",
-				ClientAppID: name + "-abcde",
-				Name:        name,
-				AppMeta:     meta,
+			cmd := &CommandCreate{
+				inputs: createInputs{
+					newAppInputs: newAppInputs{
+						Name:            "test-app",
+						Project:         "123",
+						Location:        realm.LocationVirginia,
+						DeploymentModel: realm.DeploymentModelGlobal,
+					},
+					Cluster:  tc.cluster,
+					DataLake: tc.dataLake,
+				},
 			}
-			return createdApp, nil
-		}
 
-		var importAppData interface{}
-		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
-			importAppData = appData
-			return nil
-		}
+			assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: tc.atlasClient}))
 
-		ac := mock.AtlasClient{}
-		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
-			return []atlas.Cluster{{Name: "test-cluster"}}, nil
-		}
+			localApp, err := local.LoadApp(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
+			assert.Nil(t, err)
 
-		cmd := &CommandCreate{
-			inputs: createInputs{
-				newAppInputs: newAppInputs{
-					Name:            "test-app",
-					Project:         "123",
+			assert.Equal(t, importAppData, localApp.AppData)
+			assert.Equal(t, realm.App{
+				ID:          "456",
+				GroupID:     "123",
+				Name:        "test-app",
+				ClientAppID: "test-app-abcde",
+				AppMeta: realm.AppMeta{
 					Location:        realm.LocationVirginia,
 					DeploymentModel: realm.DeploymentModelGlobal,
 				},
-				DataSource: "test-cluster"},
-		}
+			}, createdApp)
 
-		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: ac}))
+			// TODO(REALMC-8262): Investigate file path display options
+			dirLength := len(localApp.RootDir)
+			fmtStr := fmt.Sprintf("%%-%ds", dirLength)
 
-		localApp, err := local.LoadApp(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
-		assert.Nil(t, err)
+			var spaceBuffer, dashBuffer string
+			if tc.dataLake != "" {
+				spaceBuffer = "  "
```

Comment:
if we ever can figure out how to make this output deterministic (and subsequently make asserts on this output a bit easier/more straightforward), I'd like to avoid these types of `var`s in tests, mostly b/c they tend to make understanding what's being asserted a bit more difficult to parse/understand

again, I think with that filepath `display` logic below this is fine for now, just thinking out loud here...

Reply:
Agreed it's less than ideal. An Idea I just had is in the test we strip all repeating whitespace (except newlines) and dashes, then do the comparison? Since the terminal log component is handling the those two aspects which is being tested separately.

Reply:
considering we still don't even have product/design's sign off on this output style, lets avoid going to crazy trying to accommodate these tests.  lets live with the quick-and-dirty here, and see where the dust settles once people start seeing this in action